handler/handler_test.go

```diff
@@ -95,8 +200,73 @@ func TestDataImporterHandle(t *testing.T) {
 		Convey("When handle is called", func() {
 			err := batchHandler.Handle(testContext, esDestURL, testEvents)
 
-			Convey("And the error is nil", func() {
+			Convey("And the error is nil and both create and update bulk request called", func() {
 				So(err, ShouldBeNil)
+				So(count, ShouldEqual, 2)
+			})
+		})
+	})
+}
+
+//  Scenario 4
+// a) Create bulk request failed
+// b) Update bulk request not made
+func TestHandleWithCreateAndInternalServerESResponse(t *testing.T) {
+
+	var count int
+	Convey("Given a handler configured with other failed es create request", t, func() {
+
+		doFuncWithInValidResponse := func(ctx context.Context, req *http.Request) (*http.Response, error) {
+			count++
+			if count == 1 {
```

Comment:
Again for this one - there is only 1 request and response needed, so can reduce the logic in this function